File: T7/Trabajadores/src/model/Jefe.java
package model;

public class Jefe extends Trabajador implements Empleador {

    public Jefe() {
    }

    public Jefe(String nombre, String apellido, int numeroSSocial, int salario) {
        super(nombre, apellido, numeroSSocial, salario);
    }

    @Override
    public void calcularSalarioMes() {
        System.out.println(getSalario());
    }

    @Override
    public void mostrarDatos() {
        super.mostrarDatos();
    }

    @Override
    public void realizarTrabajo() {

    }
}

File: DAM2/PMDM/T1/TrabajadoresHerencia/src/model/Jefe.kt
package model

class Jefe(nombre : String, apellido : String, dni : String, salario : Double, var nResponsabilidad : Int) : Trabajador(nombre, apellido, dni, salario) {

    fun disminuirResponsabilidad(){
        if (nResponsabilidad > 0){
            nResponsabilidad--
            println("Se ha bajado el nivel de responsabilidad a $nResponsabilidad")
        }
    }

    fun aumentarResponsabilidad(){
        if (nResponsabilidad < 5){
            nResponsabilidad++
            println("Se ha aumentado el nivel de responsabilidad a $nResponsabilidad")
        }
    }

    override fun calcularSalarioNeto(): Double {
        if (nResponsabilidad < 3) {
            return salario - 0.3
        } else {
            return salario + 0.3
        }
    }

    override fun mostrarDatos() {
        super.mostrarDatos()
        println("nResponsabilidad = ${nResponsabilidad}")
    }
}